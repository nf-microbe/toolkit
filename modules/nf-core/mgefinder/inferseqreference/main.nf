process MGEFINDER_INFERSEQREFERENCE {
    tag "${meta.id}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://carsonjm/mgefinder:0.0.1' :
        'docker://carsonjm/mgefinder:0.0.1' }"

    input:
    tuple val(meta) , path(fasta)
    tuple val(meta2), path(pair)

    output:
    tuple val(meta), path("${prefix}.mgefinder.inferseq_reference.tsv") , emit: tsv
    path "versions.yml"                                                 , emit: versions

    script:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    def is_compressed = fasta.getExtension() == "gz" ? true : false
    def fasta_name = is_compressed ? fasta.getBaseName() : fasta
    """
    if [ "${is_compressed}" == "true" ]; then
        gzip -c -d ${fasta} > ${fasta_name}
    fi

    mgefinder \\
        inferseq-reference \\
        ${pair} \\
        ${fasta_name} \\
        -o ${prefix}.mgefinder.inferseq_reference.tsv \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgefinder: \$(echo \$( mgefinder --version | tail -n 1 | sed "s/MGEfinder version://" ))
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.mgefinder.inferseq_reference.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgefinder: \$(echo \$( mgefinder --version | tail -n 1 | sed "s/MGEfinder version://" ))
    END_VERSIONS
    """
}
