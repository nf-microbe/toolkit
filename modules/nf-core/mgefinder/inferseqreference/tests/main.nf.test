nextflow_process {

    name "Test Process MGEFINDER_INFERSEQREFERENCE"
    script "../main.nf"
    process "MGEFINDER_INFERSEQREFERENCE"

    tag "modules"
    tag "modules_nfcore"
    tag "mgefinder"
    tag "mgefinder/inferseqreference"

    test("fasta.gz - pair") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/nf-microbe/modules/.nf-test/tests/ba7412669bf2e3864bf37863a63e0f7b/work/e7/fb72e4eebe0fff7aaaa8c6b42283a4/test.mgefinder.pair", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/SAMEA2187099.rename.fasta", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - pair - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    []
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    []
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
