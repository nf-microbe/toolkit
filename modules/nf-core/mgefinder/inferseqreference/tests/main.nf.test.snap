{
    "fasta.gz - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f4b0fae644ad382d8293360ba4f493fc"
                ],
                "prophage": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f4b0fae644ad382d8293360ba4f493fc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-21T08:12:13.601083178"
    },
    "fasta.gz - pair": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.inferseq_reference.tsv:md5,3a447c1ab02f0847e526f3defc4c2c8a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,90f2b587faf1519f15830cb91129a9df"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.inferseq_reference.tsv:md5,3a447c1ab02f0847e526f3defc4c2c8a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,90f2b587faf1519f15830cb91129a9df"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T16:56:39.020210969"
    },
    "fasta.gz - pair - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.inferseq_reference.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,90f2b587faf1519f15830cb91129a9df"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.inferseq_reference.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,90f2b587faf1519f15830cb91129a9df"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T16:56:45.230400124"
    },
    "fasta.gz - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,8166ad81eb844e4172fd02fb2ec0db2f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f4b0fae644ad382d8293360ba4f493fc"
                ],
                "prophage": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,8166ad81eb844e4172fd02fb2ec0db2f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f4b0fae644ad382d8293360ba4f493fc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-21T08:11:53.23877017"
    }
}